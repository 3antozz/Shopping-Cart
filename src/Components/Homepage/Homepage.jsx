import styles from './Homepage.module.css'
import { Link } from 'react-router-dom'
// import PropTypes from 'prop-types';

export default function Homepage () {
    return (
        <div className={styles.homepage}>
            <h1>Welcome to <strong style={{color: "red"}}>3ANTOSHOP</strong></h1>
            <Link to='/shop'>Shop Now</Link>
        </div>
    )
}