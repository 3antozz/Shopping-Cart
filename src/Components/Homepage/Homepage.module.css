.homepage {
    font-family: 'Lobster Two';
    display: flex;
    align-items: center;
    justify-content: center;
    flex: 1;
    flex-direction: column;
    gap: 1rem;
    text-align: center;
}

.homepage h1 {
    font-size: clamp(2rem, 3.4vw, 4rem);
}


.homepage a {
    background-color: rgb(247, 96, 96);
    border-radius: 15px;
    padding: 1rem 2.5rem;
    color: white;
    text-decoration: none;
    transition: transform 200ms ease;
    font-size: clamp(2rem, 2.5vw, 2.6rem);
    font-weight: bold;
}

.homepage a:hover {
    transform: scale(1.08);
    background-color: red;
}