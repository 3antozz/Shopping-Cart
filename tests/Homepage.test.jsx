import { describe, it, expect, beforeEach } from "vitest";
import { render, screen } from "@testing-library/react";
import Homepage from "../src/Components/Homepage/Homepage.jsx";
import { MemoryRouter } from "react-router-dom";

describe("Homepage component", () => {
    // const router = createMemoryRouter(routes);
    beforeEach(() => {
        render(
            <MemoryRouter>
                <Homepage />
            </MemoryRouter>
        );
    });
    it("renders heading", () => {
        expect(screen.getByRole("heading", { name: /welcome to 3antoshop/i })).toBeInTheDocument();
    });
    it("renders button", () => {
        expect(screen.getByRole("link", { name: /shop now/i })).toBeInTheDocument();
    });
    it("navigates to the shop page on click", () => {
        expect(screen.getByRole("link", { name: /shop now/i })).toHaveAttribute("href", "/shop");
    });
    it("matches snapshot", () => {
        const {container} = render(
            <MemoryRouter>
                <Homepage />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
