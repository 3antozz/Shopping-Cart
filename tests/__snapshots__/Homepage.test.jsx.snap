// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage component > matches snapshot 1`] = `
<div>
  <div
    class="_homepage_4e2365"
  >
    <h1>
      Welcome to 
      <strong
        style="color: red;"
      >
        3ANTOSHOP
      </strong>
    </h1>
    <a
      data-discover="true"
      href="/shop"
    >
      Shop Now
    </a>
  </div>
</div>
`;
